pyquil/compatibility/v2/api/_quantum_computer.py

```diff
@@ -0,0 +1,776 @@
+        bitstring_array = self.run(executable=executable)
+        bitstring_dict = {}
+        for i, q in enumerate(qubits_to_measure):
+            bitstring_dict[q] = bitstring_array[:, i]
+        for q in set(self.qubits()) - set(qubits_to_measure):
+            bitstring_dict[q] = np.zeros(trials)
+        return bitstring_dict
+
+    def compile(
+        self,
+        program: Program,
+        to_native_gates: bool = True,
+        optimize: bool = True,
+        *,
+        protoquil: Optional[bool] = None,
+    ) -> ExecutableDesignator:
+        """
+        A high-level interface to program compilation.
+
+        Compilation currently consists of two stages. Please see the :py:class:`AbstractCompiler`
+        docs for more information. This function does all stages of compilation.
+
+        Right now both ``to_native_gates`` and ``optimize`` must be either both set or both
+        unset. More modular compilation passes may be available in the future.
+
+        Additionally, a call to compile also calls the ``reset`` method if one is running
+        on the QPU. This is a bit of a sneaky hack to guard against stale compiler connections,
+        but shouldn't result in any material hit to performance (especially when taking advantage
+        of parametric compilation for hybrid applications).
+
+        :param program: A Program
+        :param to_native_gates: Whether to compile non-native gates to native gates.
+        :param optimize: Whether to optimize the program to reduce the number of operations.
+        :param protoquil: Whether to restrict the input program to and the compiled program
+            to protoquil (executable on QPU). A value of ``None`` means defer to server.
+        :return: An executable binary suitable for passing to :py:func:`QuantumComputer.run`.
+        """
+
+        if isinstance(self.qam, QPU):
+            self.reset()
+
+        flags = [to_native_gates, optimize]
+        assert all(flags) or all(not f for f in flags), "Must turn quilc all on or all off"
+        quilc = all(flags)
+
+        if quilc:
+            nq_program = self.compiler.quil_to_native_quil(program, protoquil=protoquil)
+        else:
+            nq_program = program
+        binary = self.compiler.native_quil_to_executable(nq_program)
+        return binary
+
+    def reset(self) -> None:
+        """
+        Reset the QuantumComputer's QAM to its initial state.
+        """
+        self.qam.reset()
+
+    def __str__(self) -> str:
+        return self.name
+
+    def __repr__(self) -> str:
+        return f'QuantumComputer[name="{self.name}"]'
+
+
+def get_qc(
+    name: str,
+    *,
+    as_qvm: Optional[bool] = None,
+    noisy: Optional[bool] = None,
+    connection: Any = None,
+    compiler_timeout: float = 10,
+    client_configuration: Optional[QCSClientConfiguration] = None,
+) -> QuantumComputer:
+    """
+    Compatibility layer to build a QuantumComputer supporting the pyQuil v2 API.
```

Comment:
This isn't technically true, as the `ForestConnection`, etc still are replaced with the newer `QCSClientConfiguration`, but it's as close as possible. I don't know if there's much action here -- maybe call it a pyQuil v2-like API or something like that, here and elsewhere, if you feel it's appropriate.

Reply:
oh, good point. Fixed here and in the docs